libmcount/dynamic.c

```diff
@@ -157,73 +157,80 @@ static int find_dynamic_module(struct dl_phdr_info *info, size_t sz, void *data)
 	unsigned i;
 
 	/* TODO: support dynamic tracing for libraries */
-	if (name[0] == '\0') {
-		mdi = xzalloc(sizeof(*mdi));
+	mdi = xzalloc(sizeof(*mdi));
+
+	if (name[0] == '\0')
 		mdi->mod_name = xstrdup(read_exename());
-		mdi->base_addr = 0;
+	else
+		mdi->mod_name = xstrdup(name);
 
-		for (i = 0; i < info->dlpi_phnum; i++) {
-			if (info->dlpi_phdr[i].p_type != PT_LOAD)
-				continue;
+	mdi->base_addr = 0;
```

Comment:
This line is not needed since it's allocated by `xzalloc()` which does zero-initialization.

Reply:
removed